src/main/java/org/folio/circulation/support/http/client/QueryParameter.java

```diff
@@ -0,0 +1,8 @@
+package org.folio.circulation.support.http.client;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpRequest;
+
+public interface QueryParameter {
```

Comment:
I'd not couple the interface with vert.x framework. 

What do you think about following structure:

```
QueryParameter<T> {
  String getParameterName();
  Optional<T> getParameterValue();
}
```

And then we can introduce an intermediate class which will do the same that the `writeTo` does.

Reply:
Yes, this does mix concerns, which isn't ideal.

I thought about this kind of design. I was trying to avoid exposing the private state of the class publicly.

I might experiment with a kind of visitor to separate these concerns out, without exposing state directly.

Reply:
I've tried an experiment using a custom consumer functional interface to remove the direct coupling between these.

Please let me know what you think. Thank you for encouraging me to revisit this.